### PR TITLE
Changes for galaxies theme chooser UI

### PR DESCRIPTION
**Pull request by Stanton**
Main pulsar-level changes

* Make sure links and help text in the settings sidebar meet AA contrast
* Make sure modal footers properly clear any floats
* Add an `is-active` state for naked list items
* Stop using div.no-results as it's too specific and can't be overridden by the grid classes
* Moved grid width classes to a new `_utility` scss file which can be moved to the end of pulsar.scss and make sure grid widths are at the end of the stack (this is a breaking change)

## Breaking change checklist

The following points should be considered as an early-warning of introducing a breaking change to a Continuum platform product. This is not an exhaustive list of what constitutes a breaking change.

| Does this pull request... | Yes / No |
| ---- | ---- |
| require changes to `main.js` | ✅ No |
| require changes to `index.js` | ✅ No |
| require changes to `pulsar.scss` | 🛑 Yes |
| require changes to `gruntfile.js` | ✅ No |
| require changes to `package.json` (not including dev dependencies) | ✅ No |
| require changes to `base.html.twig` (or other core views like header/footer) | ✅ No |
| require new arguments to be passed to a js component | ✅ No |
| require markup changes to maintain functionality | ✅ No |

## Breaking change remediation

Add the following to the utilities section in `pulsar.scss` (filepaths may need modifying to match the existing imports in your file)

```
@import 'utility.grid';
```

### CODE REVIEW

File: stylesheets/_utility.utilities.scss [hunk 1]
```diff
@@ -330,6 +335,12 @@ div.no-results {
     width: auto !important;
 }
 
+%u-margin-top,
+.u-margin-top {
+margin-top: 1em;
```
Comment (jamesjacobs) [inline stylesheets/_utility.utilities.scss]: indentation

Comment (jamesjacobs) [reply stylesheets/_utility.utilities.scss]: also, shouldn't we stick to px here, rather than `em` until we move everything (or most things) to `em`?

Comment (Stanton) [reply stylesheets/_utility.utilities.scss]: Search 'em;': 169 results in 48 files ¯\_(ツ)_/¯

**Comment by Stanton** [comment]
Ready for re-review

File: stylesheets/_component.cards.scss [hunk 2]
```diff
@@ -6,6 +6,10 @@
     @include respond-min($screen-small) {
         margin-bottom: 0;
     }
+
+    &:hover {
+        background-color: #fafafa;
```
Comment (nvasileiadis) [inline stylesheets/_component.cards.scss]: Why not use the color(gray, off-white)?

File: stylesheets/_component.progress-lists.scss [hunk 3]
```diff
@@ -230,3 +230,48 @@
         z-index: 1;
     }
 }
+
+.progress-steps {
```
Comment (jamesjacobs) [inline stylesheets/_component.progress-lists.scss]: do these need a lexicon example - I'm unsure what they are for?

File: stylesheets/_utility.utilities.scss [hunk 4]
```diff
@@ -75,19 +75,24 @@ p.success {
     background-color: color(gray, off-white);
     background-image: url($image-path + 'stripes.png');
     background-repeat: repeat;
+    margin: 0 auto;
     padding: 1em 0;
     text-align: center;
-}
-
-div.no-results {
```
Comment (jamesjacobs) [inline stylesheets/_utility.utilities.scss]: What the reason behind removing this? It's used in CXM in various places and will cause regression.

File: stylesheets/_utility.utilities.scss [hunk 5]
```diff
@@ -330,6 +335,12 @@ div.no-results {
     width: auto !important;
 }
 
+%u-margin-top,
+.u-margin-top {
```
Comment (jamesjacobs) [inline stylesheets/_utility.utilities.scss]: Any reason for the `em` use here, other utility classes use `px`

File: stylesheets/_utility.grid.scss [hunk 6]
```diff
@@ -0,0 +1,46 @@
+.g-row {
```
Comment (jamesjacobs) [inline stylesheets/_utility.grid.scss]: To confirm, the grid styles are the same and have just been moved?

File: stylesheets/_structure.forms.scss [hunk 7]
```diff
@@ -54,7 +54,7 @@
 
 // Add a gutter between form labels and controls
 .control__label + .controls {
-    @include col-span($form-control-cols, false, true);
+    @include col-span($form-control-cols, false);
```
Comment (jamesjacobs) [inline stylesheets/_structure.forms.scss]: Won't this remove the gutter from all controls?

File: stylesheets/_component.modals.scss [hunk 8]
```diff
@@ -122,15 +122,10 @@
...
 .modal__footer {
     @include clear-fix; // clear it in case folks use .pull-* classes on buttons
-    padding: ($modal-inner-padding - 1) $modal-inner-padding ($modal-inner-padding - 5);
-    // text-align: right; // right align buttons
     border-top: 1px solid $modal-footer-border-color;
+    clear: both;
+    padding: ($modal-inner-padding - 1) $modal-inner-padding ($modal-inner-padding - 5);
 
-/*   // Properly space out buttons
-  .btn + .btn {
```
Comment (jamesjacobs) [inline stylesheets/_component.modals.scss]: Why remove button spacing?

File: stylesheets/_utility.utilities.scss (hunk 4 above)
Comment (Stanton) [reply stylesheets/_utility.utilities.scss]: https://github.com/jadu/pulsar/commit/da4472e0ce8cbd2392d2efd00f44559a1058f842#diff-d86f1c6d0c5c8ba3fe81b8d6c0a289a5

File: stylesheets/_utility.utilities.scss (hunk 5 above)
Comment (Stanton) [reply stylesheets/_utility.utilities.scss]: replaced with $margin-base

File: stylesheets/_utility.grid.scss (hunk 6 above)
Comment (Stanton) [reply stylesheets/_utility.grid.scss]: Correct

File: stylesheets/_structure.forms.scss (hunk 7 above)
Comment (Stanton) [reply stylesheets/_structure.forms.scss]: Not that I've noticed

File: stylesheets/_component.modals.scss (hunk 8 above)
Comment (Stanton) [reply stylesheets/_component.modals.scss]: I didn't, I removed already commented out code

File: stylesheets/_component.progress-lists.scss (hunk 3 above)
Comment (Stanton) [reply stylesheets/_component.progress-lists.scss]: removed this as it was reimplemented by Nikos as `.stepped-progressbar` (which is the style being used by CMS Galaxies Designer)

File: stylesheets/_component.modals.scss (hunk 8 above)
Comment (jamesjacobs) [reply stylesheets/_component.modals.scss]: Ah, my bad.

File: stylesheets/_structure.forms.scss (hunk 7 above)
Comment (jamesjacobs) [reply stylesheets/_structure.forms.scss]: I'd double check, that 3rd param is gutter and it's defaults to false.

File: stylesheets/_utility.utilities.scss (hunk 4 above)
Comment (jamesjacobs) [reply stylesheets/_utility.utilities.scss]: ok, can you add a note to the description to cover this as I'll need to change this in a fair few places in CXM (breaking change).

File: stylesheets/_component.progress-lists.scss (hunk 3 above)
Comment (jamesjacobs) [reply stylesheets/_component.progress-lists.scss]: 👍

**Pull request merged**